src/pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -21,175 +21,90 @@
 from torch.optim import Optimizer
 
 import pytorch_lightning as pl
+from lightning_lite.plugins import Precision as _Precision
 from lightning_lite.utilities.types import _PARAMETERS
 from pytorch_lightning.core.hooks import CheckpointHooks
 from pytorch_lightning.utilities import grad_norm, GradClipAlgorithmType
 
 
-class PrecisionPlugin(CheckpointHooks):
+class PrecisionPlugin(_Precision, CheckpointHooks):
     """Base class for all plugins handling the precision-specific parts of the training.
 
     The class attribute precision must be overwritten in child classes. The default value reflects fp32 training.
     """
 
     precision: Union[str, int] = 32
 
-    def main_params(self, optimizer: Optimizer) -> _PARAMETERS:
-        """The main params of the model.
-
-        Returns the plain model params here. Maybe different in other precision plugins.
-        """
-        for group in optimizer.param_groups:
-            yield from group["params"]
-
     def connect(
         self, model: Module, optimizers: List[Optimizer], lr_schedulers: List[Any]
     ) -> Tuple[Module, List[Optimizer], List[Any]]:
         """Connects this plugin to the accelerator and the training process."""
         return model, optimizers, lr_schedulers
 
-    def pre_backward(self, model: "pl.LightningModule", closure_loss: Tensor) -> Tensor:
-        """Run before precision plugin executes backward.
+    @contextlib.contextmanager
+    def train_step_context(self) -> Generator[None, None, None]:
```

Comment:
Moving these has messed up the diff. Is this intended?

Reply:
It is intentional, to make the order or methods consistent across plugins